File: fix_voice_ui.js
const fs = require('fs');
const filePath = process.argv[2];

// Déterminer le mode
let mode;
if (filePath.includes('avoirs')) {
  mode = 'credit_note';
} else if (filePath.includes('commandes')) {
  mode = 'order';
} else if (filePath.includes('livraisons')) {
  mode = 'delivery';
} else if (filePath.includes('acomptes')) {
  mode = 'deposit';
} else {
  mode = 'invoice';
}

const content = fs.readFileSync(filePath, 'utf8');

// Trouver et remplacer la section voice UI
const pattern = /<div className="flex flex-col items-center gap-4">[\s\S]*?<\/AnimatePresence>/;

const replacement = `<PulseVoiceRecorder
              onResult={handleVoiceResult}
              isPro={sub.canUseVoice}
              mode="${mode}"
              existingItems={items}
              sector={profile?.sector || ''}
              onClose={() => setMode('manual')}
            />`;

const newContent = content.replace(pattern, replacement);

fs.writeFileSync(filePath, newContent, 'utf8');

console.log(`Modifié ${filePath}`);

File: fix_voice_ui.py
import re
import sys

file_path = sys.argv[1]

# Déterminer le mode
if 'avoirs' in file_path:
    mode = 'credit_note'
elif 'commandes' in file_path:
    mode = 'order'
elif 'livraisons' in file_path:
    mode = 'delivery'
elif 'acomptes' in file_path:
    mode = 'deposit'
else:
    mode = 'invoice'

with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

# Trouver et remplacer la section voice UI
# Pattern: de <div className="flex flex-col items-center gap-4"> à </AnimatePresence>
pattern = r'<div className="flex flex-col items-center gap-4">.*?</AnimatePresence>'

replacement = '''<PulseVoiceRecorder
              onResult={handleVoiceResult}
              isPro={sub.canUseVoice}
              mode="''' + mode + '''"
              existingItems={items}
              sector={profile?.sector || ''}
              onClose={() => setMode('manual')}
            />'''

content_new = re.sub(pattern, replacement, content, flags=re.DOTALL)

with open(file_path, 'w', encoding='utf-8') as f:
    f.write(content_new)

print(f"Modifié {file_path}")
